python/dgl/graphbolt/impl/torch_based_feature_store.py

```diff
@@ -144,11 +145,6 @@ def update(self, value: torch.Tensor, ids: torch.Tensor = None):
             updated.
         """
         if ids is None:
-            assert self.size() == value.size()[1:], (
```

Comment:
why remove this check? not related to index_select.

Reply:
This is required if we want to be able to update the stored tensor for the inferencing use case. When inferencing, the dimensions between different layers are different, without this change, we can't update the stored feature.